### PR TITLE
terraform-compliance: init at v1.2.11

### PR DESCRIPTION
**Pull request by Charg**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have a commit per package, with the commit subject of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

you should have around 5 commits.

**Comment by kalbasit** [comment]
@Charg I hope you don't mind, but I force-pushed your branch to address the comments.

@jonringer PTAL.

File: pkgs/development/python-modules/junit-xml/default.nix [hunk 1]
```diff
@@ -19,6 +19,16 @@ buildPythonPackage rec {
...
+  #
+  #    Results (0.20s):
+  #          40 passed
+  #           1 failed
+  #             - test_junit_xml.py:255 TestSuiteTests.test_to_xml_string
+  doCheck = false;
+
   checkPhase = ''
     pytest
   '';
```
Comment (jonringer) [inline pkgs/development/python-modules/junit-xml/default.nix:34]:
```suggestion
   # some comment as to why a tests is being skipped
   checkPhase = ''
    pytest -k 'not to_xml_string'
  '';
```

File: pkgs/development/python-modules/pysingleton/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5776e7a4ba0bab26709da604f4e648c5814385fef34010723db3da0d41b0dbcc";
+  };
+
+  # No tests in the Pypi package.
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Module which provides a decorator to create thread-safe singleton classes ";
```
Comment (jonringer) [inline pkgs/development/python-modules/pysingleton/default.nix:19]: I'm getting wrapping
```suggestion
    description = "Provides a decorator to create thread-safe singleton classes";
```

File: pkgs/development/python-modules/pysingleton/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  version = "0.2.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5776e7a4ba0bab26709da604f4e648c5814385fef34010723db3da0d41b0dbcc";
+  };
+
+  # No tests in the Pypi package.
+  doCheck = false;
+
```
Comment (jonringer) [inline pkgs/development/python-modules/pysingleton/default.nix:17]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors.

**Review by jonringer — COMMENTED** [review]
overall, i would like some maintainers listed.

If tests aren't available through Pypi, then i would prefer to checkout the source.

File: pkgs/development/python-modules/pysingleton/default.nix (hunk 3 above)
Comment (kalbasit) [reply pkgs/development/python-modules/pysingleton/default.nix:19]: done

**Comment by kalbasit** [comment]
> overall, i would like some maintainers listed.
> 
> If tests aren't available through Pypi, then i would prefer to checkout the source.

@jonringer I've addressed all of your comments except for the tests for the pysingleton module. Unfortunately, the repository for this module was deleted. I did use `pythonImportsCheck` for it though.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build pythonPackages.colorful pythonPackages.pysingleton pythonPackages.tag-expressions pythonPackages.radish-bdd terraform-compliance

**Comment by kalbasit** [comment]
@GrahamcOfBorg build python3Packages.colorful python3Packages.pysingleton python3Packages.tag-expressions python3Packages.radish-bdd terraform-compliance

**Comment by kalbasit** [comment]
I force pushed to re-add @Charg as Co-Author. Not sure when/how the commit author got reset :man_facepalming: 


@GrahamcOfBorg build python3Packages.colorful python3Packages.pysingleton python3Packages.tag-expressions python3Packages.radish-bdd terraform-compliance

**Comment by kalbasit** [comment]
Merging this now, @jonringer let me know if you wanted anything different I can file a follow-up PR.

**Pull request merged**